src/types/index.d.ts

```diff
@@ -44,6 +44,11 @@ export type TableData = {
   data: any
 }
 
+export type Status = {
+  title: 'Online' | 'Offline' | 'Partially degraded' | 'Not available'
+  appearance: 'warning' | 'danger' | 'success'
+}
+
```

Comment:
We spoke about this a little offline, just to note that it would be nice to keep the English text and the styling/visuals out of the data and use something like an i18n lib and S/CSS for styling/visuals.

```javascript
// but dynamically calculating the Status. i.e. `status` here would come out as `not_available`
$t(`service.status.${status}`)
// and
:class="`status-${status}`"
```

For me at the very least taking this approach means less code to write and follow and keeps everything in the one place in the component.

Overall, right here right now, not a blocker though, just wanted to note it here.

Reply:
Have a look at the changes I added in https://github.com/kumahq/kuma-gui/pull/479/commits/c5379f44bdc52d87749dca1db3bbb9f5eee24fdb.

This isolates the mapping of "status keywords" in the status badge component and simplifies consuming code to only pass around the status keyword as it already is present on the respective data strucutres.

---

One thing I really don’t agree with is the analogy of this sort of data transformation being akin to internationalization/localization. It isn’t. We’re transforming some machine-readable data into a very few select human-readable labels and machine-readable props for the use with Kongponents. This doesn’t justify the use of an i18n library.

Reply:
Much nicer, I like that this stuff is contained in the component. Just to note: Not saying we need to pull in an i18n right here right now, but it feels to me like we'll need one at some point to do machine > human translations rather than just rolling our own (probably slightly differently each time) translation for each individual instance of this sort of stuff. Anyway one to chat about at a later date.